improver/synthetic_data/generate_metadata.py

```diff
@@ -0,0 +1,145 @@
+# LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR
+# CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF
+# SUBSTITUTE GOODS OR SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS
+# INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN
+# CONTRACT, STRICT LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE)
+# ARISING IN ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE
+# POSSIBILITY OF SUCH DAMAGE.
+"""Module to generate a metadata cube."""
+
+import numpy as np
+from iris.std_names import STD_NAMES
+
+from improver.metadata.constants.mo_attributes import MOSG_GRID_DEFINITION
+from improver.synthetic_data.set_up_test_cubes import set_up_variable_cube
+from improver.utilities.temporal import cycletime_to_datetime
+
+
+def _get_unit(name):
+    try:
+        units = STD_NAMES[name]["canonical_units"]
+        return units
+    except KeyError:
+        raise KeyError("Units of {} are not known.".format(name))
+
+
+def _create_data_array(ensemble_members, npoints, height_levels):
+    data = None
+    if ensemble_members > 1:
+        data = np.zeros((ensemble_members, npoints, npoints), dtype=int)
+    else:
+        data = np.zeros((npoints, npoints), dtype=int)
+
+    return data
+
+
+def generate_metadata(
+    name,
+    spatial_grid="latlon",
+    time="20171110T0400Z",
+    frt="20171110T0400Z",
+    ensemble_members=8,
+    attributes=None,
+    resolution=None,
+    domain_corner=None,
+    npoints=71,
+    height_levels=None,
+):
+    """ Generate a cube with metadata only.
+
+    Args:
+        name (str):
+            Output variable name.
+        spatial_grid (Optional[str]):
+            What type of x/y coordinate values to use.  Permitted values are
+            "latlon" or "equalarea".
+        time (Optional[str]):
+            Single cube validity time. Datetime string of format YYYYMMDDTHHMMZ.
+        frt (Optional[str]):
+            Single cube forecast reference time. Datetime string of format YYYYMMDDTHHMMZ.
+        ensemble_members (Optional[int]):
+            Number of ensemble members.
+        realizations (Optional[csv]):
+            CSV list of realization input data.
+        attributes (Optional[str]):
+            JSON file of additional metadata attributes.
+        resolution (Optional[float]):
+            Resolution of grid (metres or degrees).
+        domain_corner (Optional[Tuple[float, float]]):
+            Bottom left corner of grid domain (y,x) (degrees for latlon or metres for equalarea).
+        npoints (Optional[int]):
+            Number of points along a single axis.
+        height_levels (Optional[List[float]]):
+            List of altitude/pressure levels.
+
+    Returns:
+        iris.cube.Cube:
+            Output of set_up_variable_cube()
+    """
+    units = _get_unit(name)
+
+    if spatial_grid != "latlon" and spatial_grid != "equalarea":
```

Comment:
Codacy found an issue: [Consider merging these comparisons with "in" to "spatial_grid not in ('latlon', 'equalarea')"](https://app.codacy.com/manual/metoppv_tech/improver/pullRequest?prid=5894027)